src/3ds.ts

```diff
@@ -1,26 +1,33 @@
 import crypto from 'node:crypto';
 import { md5, getDataFromPathOrBuffer } from '@/util';
 
-export type CTRBOSSContainer = {
-	hash_type: number;
-	release_date: bigint;
-	iv: Buffer;
-	content_header_hash: Buffer;
-	content_header_hash_signature: Buffer;
+export type CTRPayloadContent = {
 	payload_content_header_hash: Buffer;
 	payload_content_header_hash_signature: Buffer;
 	program_id: bigint;
 	content_datatype: number;
 	ns_data_id: number;
+	version: number;
 	content: Buffer;
 }
 
+export type CTRBOSSContainer = {
+	hash_type: number;
+	serial_number: bigint;
+	iv: Buffer;
+	content_header_hash: Buffer;
+	content_header_hash_signature: Buffer;
+	payload_contents: CTRPayloadContent[];
+}
+
 export type CTRCryptoOptions = {
 	program_id?: string | number | bigint; // * Program ID and title ID are aliases
 	title_id?: string | number | bigint;   // * Program ID and title ID are aliases
-	release_date: bigint;
+	serial_number?: bigint; // * Only used in boss.encrypt()
 	content_datatype: number;
 	ns_data_id: number;
+	version: number;
+	path_or_buffer?: string | Buffer; // * Not needed in boss.encrypt()
```

Comment:
I'm not sold on the `path_or_buffer` name here, since I'm not sure it does a good job at describing what the input should *contain*? This is the payload contents so maybe it should be `payload`, or `contents`, or something?

Reply:
Renamed it to `content` for consitency with `CTRPayloadContent`